src/actions/testActions.ts

```diff
@@ -162,131 +83,102 @@ export const startTests =
+            } else if (
+                testMode === 'receiver' &&
+                channelMode === ChannelMode.single
+            ) {
+                // TODO: Figure out the importance of execution of single channel test,
+                //   this solution does not give continuous upate, but probably captures more packets.
+                // testPromise = dtm.singleChannelReceiverTest(
+                //     singleChannel,
+                //     timeout,
+                // );
+
+                // This solution works as sweep on a single channel, updates continuously
+                testPromise = dtm.sweepReceiverTest(
+                    bitpattern,
+                    length,
+                    singleChannelIndexed,
+                    singleChannelIndexed,
+                    sweepTime,
+                    timeoutms
                 );
-                dispatch(actionSucceeded(receivedChannels));
             } else {
-                logger.info(`End test failed: ${message}`);
+                testPromise = dtm.sweepReceiverTest(
+                    bitpattern,
+                    length,
+                    bleChannelsValues.indexOf(Math.min(...channelRange)),
+                    bleChannelsValues.indexOf(Math.max(...channelRange)),
+                    sweepTime,
+                    timeoutms
+                );
             }
-            dispatch(cleanup());
-        });
-
-        dispatch(startedAction(testMode));
-    };
-const cleanup = (): AppThunk<RootState, Promise<void>> => async dispatch => {
-    if (dtm.dtmTransport.port.isOpen) {
-        await dtm.dtmTransport.close();
-        dtm = null;
-    }
-    dispatch(stoppedAction());
-};
 
-export const endTests = () => {
-    logger.info('Ending test');
-    dtm.endTest();
-};
+            testPromise.then(status => {
+                if (status.type === 'error') {
+                    logger.info(`End test failed: ${status.message}`);
+                } else if (status.type === 'transmitter') {
+                    logger.info(`Transmitter test finished successfully.`);
+                    dispatch(actionSucceeded(new Array(40).fill(0)));
+                } else if (status.type === 'receiver') {
+                    dispatch(actionSucceeded(status.receivedPerChannel));
+                    logger.info(
+                        `Transmitter test finished successfully. Received ${status.receivedPerChannel.reduce(
+                            (a, b) => a + b,
+                            0
+                        )} packets.`
+                    );
+                }
+            });
 
-export const deselectDevice =
-    (): AppThunk<RootState> => (dispatch, getState) => {
-        const isRunning = getIsRunning(getState());
-        if (isRunning) {
-            endTests();
+            dispatch(startedAction(testMode));
+        } catch (e) {
+            logger.info(errorMessage);
+            dispatch(communicationError(errorMessage));
```

Comment:
Have you added proper handling in nrf-dtm-js?
I'd expect to have needed to clean up the tests (like calling the endTests below)

Reply:
Here it has no value as the only error is if serial port fails to start.

```typescript
 try {
             dtm = new DTM(port, baudRate);
             dtm.on('update', dispatch(dtmStatisticsUpdated));
             dtm.on('transport', (msg: string) => logger.debug(msg));
             dtm.on('log', (param: { message: string }) => {
                 logger.info(param.message);
             });
         } catch (e) {
             dtm.endTest();
             dispatch(cleanup());
             logger.info(errorMessage);
             dispatch(communicationError(errorMessage));
             return;
         }
```